src/dotnet/commands/dotnet-restore/RestoreCommandParser.cs

```diff
@@ -31,7 +31,28 @@ private static Option[] FullRestoreOptions()
                         "--interactive",
                         LocalizableStrings.CmdInteractiveRestoreOptionDescription,
                         Accept.NoArguments()
-                              .ForwardAs("-property:NuGetInteractive=true")) }).ToArray();
+                            .ForwardAs("-property:NuGetInteractive=true")),
+                    Create.Option(
+                        "--use-lock-file",
+                        LocalizableStrings.CmdUseLockFileOptionDescription,
+                        Accept.NoArguments()
+                            .ForwardAs("-property:RestorePackagesWithLockFile=true")),
+                    Create.Option(
+                        "--locked-mode",
+                        LocalizableStrings.CmdLockedModeOptionDescription,
+                        Accept.NoArguments()
+                            .ForwardAs("-property:RestoreLockedMode=true")),
+                    Create.Option(
+                        "--lock-file-path",
+                        LocalizableStrings.CmdLockFilePathOptionDescription,
+                        Accept.ExactlyOneArgument()
+                            .With(name: LocalizableStrings.CmdLockFilePathOption)
+                            .ForwardAsSingle(o => $"-property:NuGetLockFilePath={o.Arguments.Single()}")),
+                    Create.Option(
+                        "--reevaluate",
```

Comment:
Should this be --re-evaluate?

Reply:
@anangaur what you think? I'm fine either way

Reply:
This is hard. 

Normally a dash is a new word. Here it is a hyphenated word (in my dictionary). This makes me think naming is wrong. 

How about 

```bash
--force-evaluate
```

Reply:
ok we'll go with `--force-evaluate`

Reply:
Wouldn’t it make more sense to stick with the NuGet naming here? The underlying property is called `ReevaluateRestoreGraph`, so `--reevaluate` would be better maybe.